src/axolotl/utils/samplers/utils.py

```diff
@@ -5,12 +5,12 @@
 
 
 def get_dataset_lengths(dataset):
-    if "length" in dataset.data.column_names:
-        lengths = np.array(dataset.data.column("length"))
-    elif "position_ids" in dataset.data.column_names:
-        position_ids = dataset.data.column("position_ids")
+    if "length" in dataset.column_names:
+        lengths = np.array(dataset["length"])
+    elif "position_ids" in dataset.column_names:
+        position_ids = dataset["position_ids"]
         lengths = np.array([x[-1] + 1 for x in position_ids])
     else:
-        input_ids = dataset.data.column("input_ids")
+        input_ids = dataset["input_ids"]
         lengths = np.vectorize(len)(np.array(input_ids, dtype=object))
```

Comment:
Maybe try this change? Untested!

```suggestion
        lengths = np.array([len(seq) for seq in input_ids])
```

Since np.vectorize is not for perf but convenience (https://numpy.org/doc/stable/reference/generated/numpy.vectorize.html) , it shouldn't be much different.

Reply:
Thank you for the suggestion! The tests in `test_datasets.py` are now running fine on my machine. However, a bunch of others (e.g. in `tests/patched/test_validation.py`) are still failing for seemingly unrelated reasons. I hope this is just due to my env with WSL